config/base/services/authn-webhook/authn-webhook.yaml

```
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: authn-webhook
+  labels:
+    app.kubernetes.io/name: authn-webhook
+    app.kubernetes.io/part-of: auth-provider-zitadel
+    app.kubernetes.io/managed-by: kustomize
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: authn-webhook
+      app.kubernetes.io/part-of: auth-provider-zitadel
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: manager
+      labels:
+        app.kubernetes.io/name: authn-webhook
+        app.kubernetes.io/part-of: auth-provider-zitadel
+    spec:
+      securityContext:
+        runAsNonRoot: true
+        seccompProfile:
+          type: RuntimeDefault
+      containers:
+        - name: authn-webhook
+          image: auth-provider-zitadel:latest
+          imagePullPolicy: IfNotPresent
+          args:
+            - authn-webhook
+            - --webhook-port=8090
+            - --cert-dir=certs
+            - --cert-file=tls.crt
+            - --key-file=tls.key
+            - --zitadel-domain=$(ZITADEL_DOMAIN)
+            - --jwt-expiration=1h
+            - --metrics-bind-address=:8443
+          env:
+            - name: ZITADEL_DOMAIN
+              value: https://auth.datum.net
```

I'd recommend settings these flag values through environment variables so they can be easily patched through kustomize overlays. It's much easier to patch the `env` section of a container than trying to patch the `args` section. 

Should there also be an option for the Zitadel authentication secret used for creating the client assertion? 

```suggestion
          args:
            - authn-webhook
            - --webhook-port=$(WEBHOOK_PORT)
            - --cert-dir=$(CERT_DIR)
            - --cert-file=$(CERT_FILE)
            - --key-file=$(CERT_KEY_FILE)
            - --zitadel-domain=$(ZITADEL_DOMAIN)
            - --jwt-expiration=1h
            - --metrics-bind-address=$(METRICS_BIND_ADDR)
          env:
            - name: ZITADEL_DOMAIN
              value: https://auth.datum.net
            - name: METRICS_BIND_ADDR
              value: :8443
            - name: CERT_KEY_FILE
              value: tls.key
            - name: CERT_FILE
              value: tls.crt
            - name: CERT_DIR
              value: certs
            - name: WEBHOOK_PORT
              value: "8090"
```

---

@JoseSzycho using a file path for that would be preferred. We can use a volume and volume mount to put the file into the correct place on the filesystem. I'd recommend using a CLI flag to specify where the file path is and allow the file path to be overridden through an environment variable similar to the suggestion above.